components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/IdentityProviderSecretsProcessorImpl.java

```diff
@@ -0,0 +1,136 @@
+                        if (!StringUtils.isEmpty(prop.getValue())) {
+                            addNewIdpSecretProperty(secretName, prop);
+                            prop.setValue(buildSecretReference(secretName));
+                        }
+                    }
+                }
+            }
+        }
+
+        return identityProvider;
+    }
+
+    @Override
+    public IdentityProvider getIdpSecrets(IdentityProvider identityProvider)
+            throws SecretManagementException {
+
+        for (FederatedAuthenticatorConfig fedAuthConfig : identityProvider.getFederatedAuthenticatorConfigs()) {
+            for (Property prop : fedAuthConfig.getProperties()) {
+                if (prop.isConfidential()) {
+                    String secretName =
+                            buildSecretName(identityProvider.getId(), fedAuthConfig.getName(), prop.getName());
+                    if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
+                        ResolvedSecret resolvedSecret =
+                                secretResolveManager.getResolvedSecret(IDN_SECRET_TYPE_IDP_SECRETS, secretName);
+                        // replace secret reference with decrypted original secret
+                        prop.setValue(resolvedSecret.getResolvedSecretValue());
+                    }
+                }
+            }
+        }
+
+        return identityProvider;
+    }
+
+    @Override
+    public IdentityProvider deleteIdpSecrets(IdentityProvider identityProvider)
+            throws SecretManagementException {
+
+        for (FederatedAuthenticatorConfig fedAuthConfig : identityProvider.getFederatedAuthenticatorConfigs()) {
+            for (Property prop : fedAuthConfig.getProperties()) {
+                if (prop.isConfidential()) {
+                    String secretName =
+                            buildSecretName(identityProvider.getId(), fedAuthConfig.getName(), prop.getName());
+                    if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
+                        secretManager.deleteSecret(IDN_SECRET_TYPE_IDP_SECRETS, secretName);
+                    }
+                }
+            }
+        }
+
+        return identityProvider;
+    }
+
+    private String buildSecretName(String idpId, String fedAuthName, String propName) {
+
+        return idpId + ":" + fedAuthName + ":" + propName;
+    }
+
+    private String buildSecretReference(String secretName) throws SecretManagementException {
+
+        SecretType secretType = secretManager.getSecretType(IDN_SECRET_TYPE_IDP_SECRETS);
+        return secretType.getId() + ":" + secretName;
+    }
+
+    private void addNewIdpSecretProperty(String secretName, Property property) throws SecretManagementException {
+
+        Secret secret = new Secret();
+        secret.setSecretName(secretName);
+        secret.setSecretValue(property.getValue());
+        // secret.setDescription(property.getDescription());
```

Comment:
Do we need this commented line?

Reply:
Resolved with 5c434886aed9a689b0bfdb79e65d2758ea96abc3